src/vs/workbench/contrib/preferences/browser/settingsTree.ts

```diff
@@ -1133,6 +1133,11 @@ export class SettingArrayRenderer extends AbstractSettingRenderer implements ITr
 
 		renderArrayValidations(dataElement, template, value.map(v => v.value.data.toString()), true);
 	}
+
+	override disposeElement(_element: ITreeNode<SettingsTreeElement>, _index: number, template: IDisposableTemplate, _height: number | undefined) {
+		(template as ISettingListItemTemplate).listWidget.cancelEdit();
```

Comment:
I think a better way would be to push a disposable onto the elementDisposables that just runs a function that calls cancelEdit (see the `toDisposable` helper) which I think is nicer than overriding a dispose method to do something that is not disposing

Reply:
That's a lot better, implemented